gobblin-docs/user-guide/Docker-Integration.md

```diff
@@ -12,74 +12,104 @@ For more information on Docker, including how to install it, check out the docum
+
+* Second, in the same folder of the yml file, create a [hadoop.env](https://github.com/apache/incubator-gobblin/blob/master/gobblin-docker/gobblin-recipes/kafka-hdfs/hadoop.env) file to specify all HDFS related config(copy the content into your .env file).
+
+* Open a terminal in the same folder, pull and run these docker services:
+
+    `docker-compose -f ./docker-compose.yml pull`
+
+    `docker-compose -f ./docker-compose.yml up`
+    
+    Here we expose Zookeeper at port 2128, Kafka at 9092 with an auto created Kafka topic “test”. All hadoop related configs are stated in the .env file.
+
+* You should see all services running. Now we can push some events into the Kafka topic. Open a terminal from [docker desktop](https://docs.docker.com/desktop/dashboard/) dashboard or [docker exec](https://docs.docker.com/engine/reference/commandline/exec/) to interact with Kafka. Inside the Kafka container terminal:
+
+    `cd /opt/kafka`
+
+    `./bin/kafka-console-producer.sh --broker-list kafka:9092 --topic test`
+
+    You can type messages for the topic “test”, and press ctrl+c to exit.
+
+* Put the [kafka-hdfs.pull](https://github.com/apache/incubator-gobblin/blob/master/gobblin-example/src/main/resources/kafka-hdfs.pull) in ${LOCAL_JOB_DIR}, so that the Gobblin daemon will pick up this job and write the result to HDFS. You will see the Gobblin daemon pick up the job.
+
+After the job finished, open a terminal in the HDFS namenode container:
+
+`hadoop fs -ls /gobblintest/job-output/test/`
+
+You will see the result file in this HDFS folder. You can use this command to verify the content in the text file:
+
+`hadoop fs -cat /gobblintest/job-output/test/<output_file.txt>`
+
+# Run Gobblin as a Service
+
+The goal of GaaS(Gobblin as a Service) is to enable a self service so that different users can automatically provision and execute various supported Gobblin applications limiting the need for development and operation teams to be involved during the provisioning process. You can take a look at our [design detail](https://cwiki.apache.org/confluence/display/GOBBLIN/Gobblin+as+a+Service).
+
+We use the same docker image as discussed above. 
+
+### Set working directory
+
+Similar to standalone working directory settings:
+
+`export GAAS_JOB_DIR=<gaas_gobblin_directory>`
 
-Running the `gobblin-standalone` image requires taking the following steps:
+`export LOCAL_DATAPACK_DIR=<local_directory_of_templateUris>`
 
-* Download the images from the `gobblin/gobblin-standalone` repository
+### Start Gobblin as a Service
 
-```
-docker pull gobblin/gobblin-standalone:ubuntu-gobblin-latest
-```
+Run these commands to start the docker image:
 
-* Run the `gobblin/gobblin-standalone:ubuntu-gobblin-latest` image in a Docker container
+`docker pull apache/gobblin:latest`
```

Comment:
So this is sort of awkward but the apache image actually doesn't support GaaS right now. I have a PR I will open later this week to support multiple execution modes with the image but for the meantime could we use our own image at `gobblin/gobblin-service` for GaaS specifically? I can change this doc in my PR to update the image to keep the documentation up to date.